core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -523,6 +523,19 @@ default Fingerprint getParsedFingerprint(String path) {
    */
   UfsStatus getStatus(String path) throws IOException;
 
+  /**
+   * Gets the file or directory status. The caller does not need to know if the path is a file or
+   * directory. This method will determine the path type, and will return the appropriate status.
+   *
+   * @param path the path to get the status
+   * @param options method options
+   * @return the file or directory status
+   * @throws FileNotFoundException when the path does not exist
+   */
+  default UfsStatus getStatus(String path, GetFileStatusOptions options) throws IOException {
```

Comment:
Ideally, we need to remove this default implementation and have the default implementation at the method above: getStatus(String path). Have such change to limit the scope of this cherry-pick PR otherwise it would change a lot of ufs.